pisound-btn/pisound-btn.c

```diff
@@ -79,27 +75,29 @@ static const char *const PISOUND_VERSION_FILE      = "/sys/kernel/pisound/versio
 
 static const char *const UPDATE_CHECK_DISABLE_FILE = BASE_PISOUND_DIR "/disable_update_check"; // If the file exists, the update check will be disabled.
 
-static const char *const DEFAULT_DOWN              = BASE_SCRIPTS_DIR "/system/down.sh";
-static const char *const DEFAULT_UP                = BASE_SCRIPTS_DIR "/system/up.sh";
-
 static const char *const DEFAULT_CLICK_1           = BASE_SCRIPTS_DIR "/start_puredata.sh";
 static const char *const DEFAULT_CLICK_2           = BASE_SCRIPTS_DIR "/stop_puredata.sh";
 static const char *const DEFAULT_CLICK_3           = BASE_SCRIPTS_DIR "/toggle_wifi_hotspot.sh";
 
-// Receives 'times clicked' argument.
-static const char *const DEFAULT_CLICK_OTHER       = BASE_SCRIPTS_DIR "/do_nothing.sh";
-
 // Receive 'held after n clicks' and 'time held' arguments.
-static const char *const DEFAULT_HOLD_1S           = BASE_SCRIPTS_DIR "/do_nothing.sh";
 static const char *const DEFAULT_HOLD_3S           = BASE_SCRIPTS_DIR "/toggle_bt_discoverable.sh";
 static const char *const DEFAULT_HOLD_5S           = BASE_SCRIPTS_DIR "/shutdown.sh";
-static const char *const DEFAULT_HOLD_OTHER        = BASE_SCRIPTS_DIR "/do_nothing.sh";
 
 // Arbitrarily chosen limit.
 enum { MAX_PATH_LENGTH = 4096 };
 
 static char g_config_path[MAX_PATH_LENGTH+1]  = "/etc/pisound.conf";
 
+// must hold the CLICK_OTHER_VALUE_NAME or HOLD_OTHER_VALUE_NAME values
+#define ACTION_NAME_SIZE 11
+// printf( CLICK_%u", ABSOLUTE_MAX_CLICK) must fit in ACTION_NAME_SIZE
+#define ABSOLUTE_MAX_CLICK 99
+// printf( HOLD_%uS", ABSOLUTE_MAX_HOLD) must fit in ACTION_NAME_SIZE
+#define ABSOLUTE_MAX_HOLD 99
+// convert ticks to seconds so the [0,2) == 1, [2,4) == 3, [4,6) == 5
+#define TICK_2_SECONDS(x) (x/1000 + (x%2))
+static char ACTION_NAME[ACTION_NAME_SIZE+1];
```

Comment:
Please don't use a global variable here for this, it limits the flexibility of the code - prefer copying into stack variable like get_action_script does.